docs/design/app/workflows.md

```diff
@@ -1,3 +1,35 @@
+### Workflow Onset
+
+At the beginning of each workflow, a `Transaction` has to be parsed and validated.
+As these steps are the same for all workflows, this functionality has been extracted and made available independently.
+All related classes can be found in the package `com.hedera.node.app.workflows.onset`.
+Details about the required pre-checks can be found [here](transaction-prechecks.md).
+
+### Ingest Workflow
+
+The package `com.hedera.node.app.workflows.ingest` contains the ingest workflow. A rough overview can be seen in the diagram below.
+
+![Diagram of ingest workflow](images/Ingest%20Workflow.png)
```

Comment:
I think box 5 in the diagram should refer to the latest immutable state, rather than the latest signed state.

Reply:
I'm not sure if we have the throttle check in the right place. We need to do something special here I think for privileged accounts, and move the throttle towards to front.

Reply:
Modified it according to our discussion